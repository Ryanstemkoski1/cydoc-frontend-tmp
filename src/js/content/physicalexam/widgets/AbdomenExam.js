import React, {Component} from 'react'
import { Table} from 'semantic-ui-react'
import AbdomenExamButtons from './AbdomenExamButtons'
import HPIContext from '../../../contexts/HPIContext'

export default class AbdomenExam extends Component {

    static contextType = HPIContext 

    render = () => {
        var ab_map = []
        const ab_quadrants = Object.keys(this.context["Physical Exam"].widgets["Abdomen"])
        for (var ab_index = 0; ab_index < ab_quadrants.length/2; ab_index ++) {
            ab_map.push(
                <Table.Row key={ab_quadrants[2*ab_index]}>
                    <Table.Cell> <div style={{marginBottom: 5}}> {ab_quadrants[2*ab_index]} </div>
                        <AbdomenExamButtons key={ab_quadrants[2*ab_index]} ab_quadrant={ab_quadrants[2*ab_index]}/> 
                    </Table.Cell>
                    <Table.Cell> <div style={{marginBottom: 5}}> {ab_quadrants[2*ab_index+1]}  </div>
                        <AbdomenExamButtons key={ab_quadrants[2*ab_index+1]} ab_quadrant={ab_quadrants[2*ab_index+1]}/> 
                    </Table.Cell>
                </Table.Row>
            )
        }
        return (
            <Table celled collapsing>
              <Table.Body>
                {ab_map}
              </Table.Body>
              </Table>
              )
        }
    }
