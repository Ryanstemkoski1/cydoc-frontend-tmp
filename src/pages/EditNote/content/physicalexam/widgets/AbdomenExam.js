import React, { Component } from 'react';
import { Table } from 'semantic-ui-react';
import AbdomenExamButtons from './AbdomenExamButtons';
import HPIContext from 'contexts/HPIContext.js';
import './TableCSS.css';

export default class AbdomenExam extends Component {
    static contextType = HPIContext;

    render = () => {
        let ab_map = [];
        const ab_quadrants = Object.keys(
            this.context['Physical Exam'].widgets['Abdomen']
        );
        for (let ab_index = 0; ab_index < ab_quadrants.length / 2; ab_index++) {
            ab_map.push(
                <Table.Row key={ab_quadrants[2 * ab_index]}>
                    <Table.Cell>
                        {' '}
                        <div style={{ marginBottom: 5 }}>
                            {' '}
                            {ab_quadrants[2 * ab_index]}{' '}
                        </div>
                        <AbdomenExamButtons
                            key={ab_quadrants[2 * ab_index]}
                            ab_quadrant={ab_quadrants[2 * ab_index]}
                        />
                    </Table.Cell>
                    <Table.Cell>
                        {' '}
                        <div style={{ marginBottom: 5 }}>
                            {' '}
                            {ab_quadrants[2 * ab_index + 1]}{' '}
                        </div>
                        <AbdomenExamButtons
                            key={ab_quadrants[2 * ab_index + 1]}
                            ab_quadrant={ab_quadrants[2 * ab_index + 1]}
                        />
                    </Table.Cell>
                </Table.Row>
            );
        }
        return (
            <Table celled unstackable>
                <Table.Body>{ab_map}</Table.Body>
            </Table>
        );
    };
}
