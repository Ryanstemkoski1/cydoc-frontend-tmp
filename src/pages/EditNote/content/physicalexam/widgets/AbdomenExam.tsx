import React, { Component } from 'react';
import { Table } from 'semantic-ui-react';
import AbdomenExamButtons from './AbdomenExamButtons';
import './TableCSS.css';

export default class AbdomenExam extends Component {
    render = () => {
        return (
            <div>
                <Table celled unstackable>
                    <Table.Body>
                        <Table.Row key={'upperQuadrant'}>
                            <Table.Cell>
                                {' '}
                                <div> Right Upper Quadrant</div>
                                <AbdomenExamButtons
                                    key={'rightUpperQuadrant'}
                                    abdomenQuadrant={'rightUpperQuadrant'}
                                />
                            </Table.Cell>
                            <Table.Cell>
                                {' '}
                                <div> Left Upper Quadrant</div>
                                <AbdomenExamButtons
                                    key={'leftUpperQuadrant'}
                                    abdomenQuadrant={'leftUpperQuadrant'}
                                />
                            </Table.Cell>
                        </Table.Row>
                        <Table.Row key={'lowerQuadrant'}>
                            <Table.Cell>
                                {' '}
                                <div> Right Lower Quadrant</div>
                                <AbdomenExamButtons
                                    key={'rightLowerQuadrant'}
                                    abdomenQuadrant={'rightLowerQuadrant'}
                                />
                            </Table.Cell>
                            <Table.Cell>
                                {' '}
                                <div> Left Lower Quadrant</div>
                                <AbdomenExamButtons
                                    key={'leftLowerQuadrant'}
                                    abdomenQuadrant={'leftLowerQuadrant'}
                                />
                            </Table.Cell>
                        </Table.Row>
                    </Table.Body>
                </Table>
            </div>
        );
    };
}
