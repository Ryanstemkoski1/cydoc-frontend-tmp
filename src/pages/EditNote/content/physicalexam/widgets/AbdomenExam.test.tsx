import { shallow } from 'enzyme';
import React from 'react';
import AbdomenExam from './AbdomenExam';

describe('AbdomenExam', () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<AbdomenExam />);
        expect(wrapper).toBeTruthy();
    });

    it('matches snapshot', () => {
        const wrapper = shallow(<AbdomenExam />);
        expect(wrapper).toMatchSnapshot();
    });
});
