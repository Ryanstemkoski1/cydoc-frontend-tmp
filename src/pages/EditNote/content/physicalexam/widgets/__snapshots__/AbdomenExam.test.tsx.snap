// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbdomenExam matches snapshot 1`] = `
<div>
  <Table
    as="table"
    celled={true}
    unstackable={true}
  >
    <TableBody
      as="tbody"
    >
      <TableRow
        as="tr"
        cellAs="td"
        key="upperQuadrant"
      >
        <TableCell
          as="td"
        >
           
          <div>
             
            Right Upper Quadrant
          </div>
          <Connect(AbdomenExamButtons)
            abdomenQuadrant="rightUpperQuadrant"
            key="rightUpperQuadrant"
          />
        </TableCell>
        <TableCell
          as="td"
        >
           
          <div>
             
            Left Upper Quadrant
          </div>
          <Connect(AbdomenExamButtons)
            abdomenQuadrant="leftUpperQuadrant"
            key="leftUpperQuadrant"
          />
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td"
        key="lowerQuadrant"
      >
        <TableCell
          as="td"
        >
           
          <div>
             
            Right Lower Quadrant
          </div>
          <Connect(AbdomenExamButtons)
            abdomenQuadrant="rightLowerQuadrant"
            key="rightLowerQuadrant"
          />
        </TableCell>
        <TableCell
          as="td"
        >
           
          <div>
             
            Left Lower Quadrant
          </div>
          <Connect(AbdomenExamButtons)
            abdomenQuadrant="leftLowerQuadrant"
            key="leftLowerQuadrant"
          />
        </TableCell>
      </TableRow>
    </TableBody>
  </Table>
</div>
`;
